### PR TITLE
Add instance tags

### PR DESCRIPTION
**Pull request by kelibezhani**
Add arbitrary tags to instances from `tags_default` and hostclass `tags` in `disco_aws.ini`.

Read `environment_class` tag from VPC and ad it to instances.

### CODE REVIEW

File: disco_aws_automation/disco_aws.py [hunk 1]
```diff
@@ -394,6 +394,24 @@ def provision(self, ami, hostclass=None, owner=None, instance_type=None, monitor
...
         elb = self.update_elb(hostclass, update_autoscaling=False, testing=testing)
 
+        tags = {"hostclass": hostclass,
+                "owner": user_data["owner"],
+                "environment": self.environment_name,
+                "chaos": chaos,
+                "is_testing": "1" if testing else "0"}
+
+        if self.vpc.environment_class:
+            tags['environment_class'] = self.vpc.environment_class
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_aws.py]: Are sandboxes correctly tagged everywhere? Maybe this should default to `development`.

File: disco_aws_automation/disco_aws.py [hunk 2]
```diff
@@ -394,6 +394,24 @@ def provision(self, ami, hostclass=None, owner=None, instance_type=None, monitor
...
+
+        hostclass_option_tags = self.hostclass_option_default(hostclass, "tags")
+        if hostclass_option_tags:
+            hostclass_tags = {k.strip(): v.strip() for k, v in
+                              [tag.split(':') for tag in hostclass_option_tags.split(',')]}
+        else:
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_aws.py]: As it stands, this else doesn't seem very necessary. You could just do the update after formulating the `hostclass_tags`.

File: disco_aws_automation/disco_aws.py [hunk 3]
```diff
@@ -394,6 +394,24 @@ def provision(self, ami, hostclass=None, owner=None, instance_type=None, monitor
...
+                "environment": self.environment_name,
+                "chaos": chaos,
+                "is_testing": "1" if testing else "0"}
+
+        if self.vpc.environment_class:
+            tags['environment_class'] = self.vpc.environment_class
+
+        hostclass_option_tags = self.hostclass_option_default(hostclass, "tags")
+        if hostclass_option_tags:
+            hostclass_tags = {k.strip(): v.strip() for k, v in
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_aws.py]: The whitespacing here makes this a little hard to follow. Maybe something like this is clearer?

```python
for tag in hostclass_option_tags.split(','):
    for k, v in tag.split(':'):
        tags[k.strip()] = v.strip()
```

We lose the comprehension, but I think it made things a little harder to see, and we do lose all the conditional logic and the update line.

File: disco_aws_automation/disco_aws.py [hunk 4]
```diff
@@ -394,6 +394,24 @@ def provision(self, ami, hostclass=None, owner=None, instance_type=None, monitor
 
         elb = self.update_elb(hostclass, update_autoscaling=False, testing=testing)
 
+        tags = {"hostclass": hostclass,
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_aws.py]:
```suggestion
        tags = {
            "hostclass": hostclass,
```
A bit of a nitpick, but I don't like the whitespacing that this generates when we have the curly bracket and the first item on the same line. I think the above is easier to follow.

File: disco_aws_automation/disco_aws.py (hunk 1 above)
Comment (kelibezhani) [reply disco_aws_automation/disco_aws.py]: If VPCs are not tagged with `environment_class`, this tag is not added at all for the hostclass. I did not have a real tag value to default to, but I can default it to `development`.
